### DISCUSSION TITLE
Allow Defined Metrics to be Selectable in Filter (not custom SQL)

### DISCUSSION THREAD

**Discussion by jamesrayoub** [apache/superset#18493] [Ideas]
**Is your feature request related to a problem? Please describe.**
It's frustrating to have to rehash metric logic in the custom SQL filter when it already exists in the semantic layer. If I make a change to a metric in the semantic layer, I also need to remember to fix all instances where I use that metric as a filter

**Describe the solution you'd like**
Instead of having to paste the metric SQL into the custom SQL filter tab, the user could either select the metric like they can in the 'simple' tab or, perhaps there's a way to reference the metric in the custom SQL tab so that it will compile the query using the defined metric

**Describe alternatives you've considered**
Simple copy and paste from the metric definition

**Comment by rumbin** [comment]
Great idea. The metric would then of course create a HAVING filter.
